ideascube/library/tests/test_views.py

```diff
@@ -108,6 +114,26 @@ def test_book_detail_page_does_not_list_specimens_to_non_staff(app, book):
     resp.mustcontain(no=[specimen.barcode])
 
 
+def test_book_detail_page_shows_warning_message_to_staff_if_no_specimens(staffapp, book):
+    resp = staffapp.get(reverse('library:book_detail', kwargs={'pk': book.pk}))
+    resp.mustcontain("Please add a specimen, or the book won&#39;t be available for the users")
+
+
+def test_book_detail_page_does_not_show_warning_message_to_user_if_no_specimens(app, book):
```

Comment:
Not a big deal imho.
We do not show the books without specimen in the list to normal users, because we don't want to polute their list, and we do not want them to think they can read those books, because we actually don't have them.
But if they manage to get on the detail page anyway, that's not a "problem" (even if this is something that should not happen, i.e. we should not have links pointing here).